samcli/hook_packages/terraform/copy_terraform_built_artifacts.py

```diff
@@ -249,6 +251,11 @@ def cli_exit():
         LOG.error(ex.message, exc_info=True)
         cli_exit()
 
+    if not isinstance(extracted_attribute_path, string_types):
+        LOG.error("Expected the extracted attribute to be a string")
+        cli_exit()
+    extracted_attribute_path = str(extracted_attribute_path)
```

Comment:
mypy was complaining about this
```
samcli/hook_packages/terraform/copy_terraform_built_artifacts.py:261: error: Argument 2 to "join" has incompatible type "Union[Any, None, str]"; expected "Union[str, _PathLike[str]]"
samcli/hook_packages/terraform/copy_terraform_built_artifacts.py:262: error: Argument 1 to "isabs" has incompatible type "Union[Any, None, str]"; expected "Union[str, bytes, _PathLike[str], _PathLike[bytes]]"
samcli/hook_packages/terraform/copy_terraform_built_artifacts.py:263: error: Argument 1 to "abspath" has incompatible type "Union[Any, None, str]"; expected "_PathLike[Any]"
```

Reply:
@sriram-mv this check is what I meant with my comment here https://github.com/aws/aws-sam-cli/pull/4117#discussion_r950594500
😅